src/lib/components/fund/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Flex, Text } from "@chakra-ui/react";
+import type { Control, UseFormSetValue } from "react-hook-form";
+import { useWatch } from "react-hook-form";
+
+import { SelectInput } from "lib/components/forms";
+
+import { JsonFund } from "./jsonFund";
+import { SelectFund } from "./selectFund";
+import type { AttachFundsState } from "./types";
+import { AttachFundsType } from "./types";
+
+interface AttachFundContentProps {
+  control: Control<AttachFundsState>;
+  setValue: UseFormSetValue<AttachFundsState>;
+}
+
+const attachFundOptions = [
+  {
+    label: "No funds attached",
+    value: AttachFundsType.ATTACH_FUNDS_NULL,
+    disabled: false,
+  },
+  {
+    label: "Select asset and fill amount",
+    value: AttachFundsType.ATTACH_FUNDS_SELECT,
+    disabled: false,
+  },
+  {
+    label: "Provide JSON Asset List",
+    value: AttachFundsType.ATTACH_FUNDS_JSON,
+    disabled: false,
+  },
+];
+
+const AttachFundContent = ({ control, setValue }: AttachFundContentProps) => {
+  const [assetsSelect, assetsJson, attachFundOption] = useWatch({
+    control,
+    name: ["assetsSelect", "assetsJson", "attachFundOption"],
```

Comment:
IMO, let's rename and declare like this for preventing mismatch string in the future (if anyone change it.)
```tsx
// data.ts in /funds
// remark: use as a form key
const SELECTED_ASSETS = "selectedAssets";
const ASSETS_JSON_STR = "assetsJsonStr";
const ATTACH_FUNDS_OPTION = "attachFundOption";
```